schema/query/predicate.go

```diff
@@ -545,3 +547,69 @@ func (e *Regex) Prepare(validator schema.Validator) error {
 func (e Regex) String() string {
 	return quoteField(e.Field) + ": {" + opRegex + ": " + valueString(e.Value) + "}"
 }
+
+// ElemMatch matches object values specified in an array.
+type ElemMatch struct {
+	Field string
+	Exps  []Expression
+}
+
+// Match implements Expression interface.
+func (e ElemMatch) Match(payload map[string]interface{}) bool {
+	value := getField(payload, e.Field)
+
+	var arr []interface{}
+	var ok bool
+	if arr, ok = value.([]interface{}); !ok {
+		return false
+	}
+
+	if e.Exps == nil || len(e.Exps) == 0 {
+		// nil or empty predicates will not match
+		return false
+	}
+	for _, val := range arr {
+		if v, ok := val.(map[string]interface{}); ok {
+			match := true
+			for _, subQuery := range e.Exps {
+				if !subQuery.Match(v) {
+					match = false
+				}
+			}
+			if match {
+				return true
+			}
```

Comment:
Does it work to just do this?

```go
p := Predicate: Predicate(e.Exps)
for _, val := range arr {
    if p.Match(val) {
        return true
    }
}
```

Reply:
Well `p.Match` needs `map[string]interface{}` and plain `interface{}` will not work. Apart from that good catch +1.